syntax = "proto3";

service Manager {
    rpc AddPessoa(Pessoa) returns (BooleanReply) {}
    rpc DelPessoa(NomePessoa) returns (BooleanReply) {}
    rpc BuscaPessoa(NomePessoa) returns (stream Pessoa) {}
    rpc ListaAgenda(ContactsRequest) returns (stream Pessoa) {}
}

message Pessoa {
    string nome = 1;
    string celular = 2;
    string telefone = 3;
}

message NomePessoa {
    string nome = 1;
}

message BooleanReply {
    bool reply = 1;
}

message ContactsRequest {
    bool ListaAgenda = 1;
}